workerbase/lib/build_crosscompiler.sh

```diff
@@ -342,9 +384,21 @@ install_gcc()
         GCC_CONF_ARGS="${GCC_CONF_ARGS} --enable-languages=c,c++,fortran,objc,obj-c++"
     fi
 
-    if [[ "${target}" == *linux* ]]; then
+    if [[ "${target}" == *linux* || "${target}" == *freebsd* ]]; then
         GCC_CONF_ARGS="${GCC_CONF_ARGS} --enable-languages=c,c++,fortran"
-        GCC_CONF_ARGS="${GCC_CONF_ARGS} --with-sysroot=$(get_sysroot)" 
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --with-sysroot=$(get_sysroot)"
+    fi
+
+    # Some more FreeBSD-specific settings
+    if [[ "${target}" == *freebsd* ]]; then
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --without-headers"
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --with-gnu-as"
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --with-gnu-ld"
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --disable-nls"
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --enable-libssp"
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --enable-ld"
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --disable-libitm"
+        GCC_CONF_ARGS="${GCC_CONF_ARGS} --disable-libgomp"
```

Comment:
Do you know whether these are actually necessary? I copied them from somewhere originally but I can't recall where.

Reply:
I'm not quite sure about every option, but here is what I have (experience + search :P):

* `--without-headers` => for building cross compiler, might be fine to disable it, since we use Clang ?
* `--with-gnu-as` => assume using GNU assembler
* `--with-gnu-ld` => assume using GNU linker
* `--disable-nls` => Native Language Support, should be fine
* `--enable-libssp` => stack smashing protection
* `--enable-ld` => for binutils, paired with --enable-gold
* `--disable-libitm` => Transactional Memory support, probably used in WIP C++ spec ? not quiet sure
* `--disable-libgomp` => OpenMP support, some project might use it to leverage multi-core CPU, but LLVM has another one

Ref: https://gcc.gnu.org/install/configure.html#with-gnu-as

Reply:
Since we're not installing `gcc` for FreeBSD; I assume we can just remove these parts?